features/include/pcl/features/impl/good.hpp

```diff
@@ -0,0 +1,666 @@
+{
+  for (size_t i = 0 ; i < histogram_2D.size(); i++)
+  {   
+    for (size_t j = 0; j < histogram_2D.at(i).size(); j++)
+    {
+      histogram.push_back (histogram_2D.at(i).at(j));
+    }
+  }
+}
+//////////////////////////////////////////////////////////////////////////////////////////////
+
+template <typename PointInT, int NumberOfBins>
+void
+pcl::GOODEstimation<PointInT, NumberOfBins>::signDisambiguationXAxis (PointCloudIn  XoZ_projected_view, float threshold, int &sign )
+{
+  unsigned Xpositive = 0; 
+  unsigned Xnegative = 0; 
+  unsigned threshold_overall = trunc (XoZ_projected_view -> points.size() / 10);
+
+  //XoZ page
+  for (size_t i = 0; i < XoZ_projected_view -> points.size(); i++)
+  {	
+    if (XoZ_projected_view->points.at(i).x > threshold)
+    {
+      Xpositive ++;
+    }
+    else if (XoZ_projected_view->points.at(i).x < - threshold)
+    {
+      Xnegative ++;
+    }	  
+  }  
+  if ((Xpositive < Xnegative) and (Xnegative - Xpositive >= threshold_overall)) 
+  {
+    sign = -1;
+  }
+  else 
+  {
+    sign = 1;
+  }		  
+}
+//////////////////////////////////////////////////////////////////////////////////////////////
+
+template <typename PointInT, int NumberOfBins>
+void 
+pcl::GOODEstimation<PointInT, NumberOfBins>::signDisambiguationYAxis (PointCloudIn  YoZ_projected_view, float threshold, int &sign )
+{
+  unsigned int Ypositive =0; 
+  unsigned int Ynegative =0; 
+  unsigned int threshold_overall = trunc (YoZ_projected_view -> points.size() / 10);
+  //YoZ page
+  for (size_t i = 0; i < YoZ_projected_view -> points.size(); i++)
+  {	
+    if (YoZ_projected_view->points.at(i).y > threshold)
+    {
+      Ypositive ++;
+    }
+    else if (YoZ_projected_view->points.at(i).y < - threshold)
+    {
+      Ynegative ++;
+    }
+  }  
+  if ((Ypositive < Ynegative) and (Ynegative - Ypositive >= threshold_overall))
+  {
+    sign = -1;
+  }
+  else 
+  {
+    sign = 1;
+  }		  
+}
+//////////////////////////////////////////////////////////////////////////////////////////////
+ 
+ 
+template <typename PointInT, int NumberOfBins>
+void
+pcl::GOODEstimation<PointInT, NumberOfBins>::create2DHistogramFromYOZProjection (PointCloudIn  YOZ_projected_view,
+			double largest_side, unsigned int number_of_bins, int sign, std::vector<std::vector<unsigned int> > &YOZ_histogram)
+{
+  double y = largest_side / 2; 
+  double z = largest_side / 2; 
+  double interval_y = largest_side / number_of_bins; 
+  double interval_z = largest_side / number_of_bins; 
+  for (size_t i=0; i < YOZ_projected_view->points.size(); i++)
+  {
+    pcl::PointXYZ p;
+    p.y = sign * YOZ_projected_view->points.at(i).y + y;
+    p.z = YOZ_projected_view ->points.at(i).z + z;
+    
+    //if adaptive_support_lenght parameter == false, some points might be projected outside of the plane, we must discard them.
+    if ((trunc(p.y / interval_y) < YOZ_histogram.size()) and (trunc(p.z / interval_z) < YOZ_histogram.at(0).size())
```

Comment:
Please (a) do not use `and` for logical operations, use `&&` instead; (b) qualify `trunc` with `std` namespace, i.e. `std::trunc`, otherwise MSVC users will be sad.